packages/node/src/indexer/fetch.service.ts

```diff
@@ -34,18 +35,28 @@ import {
 const BLOCK_TIME_VARIANCE = 5000;
 
 const INTERVAL_PERCENT = 0.9;
-const QUERY_ADDRESS_LIMIT = 50;
 
 function eventFilterToQueryEntry(
   filter: EthereumLogFilter,
   dsOptions: SubqlEthereumProcessorOptions | SubqlEthereumProcessorOptions[],
 ): DictionaryQueryEntry {
+  const { argv } = yargsOptions;
+
   const conditions: DictionaryQueryCondition[] = [];
 
   if (Array.isArray(dsOptions)) {
     const addresses = dsOptions.map((option) => option.address).filter(Boolean);
 
-    if (addresses.length !== 0 && addresses.length <= QUERY_ADDRESS_LIMIT) {
+    if (addresses.length > argv['query-address-limit']) {
```

Comment:
I would move `argv['query-address-limit']` so we only have one occurrence rather than 3